pkg/mpc-examples/contracts/03-ebu-rebalancer/EBURebalancer.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-pool-utils/contracts/lib/ComposablePoolLib.sol";
+
+contract EBURebalancer {
+    IVault private immutable _vault;
+    bytes32 private immutable _poolId;
+    IManagedPool private immutable _pool;
```

Comment:
ditch `_pool` to save on storage; use `_getPoolFromId(getPoolId())` instead

Reply:
We use ```_pool``` in 4 functions within this contract. Isn't it a cleaner solution to just use ```_pool```. What is the determining factor in when we want to store a reusable variable vs constantly using its internal call